lib/attack.go

```diff
@@ -313,14 +243,16 @@ func (a *Attacker) Attack(tr Targeter, r Rater, name string) <-chan *Result {
 		defer close(ticks)
 		began := time.Now()
 		for {
-			wait := r.Wait(began, time.Now())
-			if wait < 0 {
+			elapsed := time.Since(began)
+			wait, stop := p.Pace(elapsed, count)
+			if stop || du > 0 && elapsed > du {
```

Comment:
I think it would be clearer if we check `du > 0` just once - the very first thing we do in `Attack` function - and return if this condition is not satisfied.